### PR TITLE
Embed `UserNSSandbox_jll` source into `deps/userns_sandbox.c`

### PR DESCRIPTION
**Pull request by staticfloat**
Also add CI configuration to test with local build of `sandbox` executable.

### CODE REVIEW

**Comment by staticfloat** [comment]
@giordano does this look about right to you?

**Comment by staticfloat** [comment]
@maleadt Your wish is my command; this should give us some CI coverage for changes to `sandbox.c`.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .vscode/
-Manifest.toml
\ No newline at end of file
+Manifest.toml
+LocalPreferences.toml
```
Comment (giordano) [inline .gitignore:3]:
```suggestion
LocalPreferences.toml

```

File: deps/build_local_sandbox.jl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+sandbox_src = joinpath(@__DIR__, "userns_sandbox.c")
+run(`cc -std=c99 -O2 -static -static-libgcc -g -o $(sandbox_path) $(sandbox_src)`)
+
+# Tell UserNSSandbox_jll to load our `sandbox` instead of the default artifact one
+set_preferences!(
+    joinpath(dirname(@__DIR__), "LocalPreferences.toml"),
+    "UserNSSandbox_jll",
+    "sandbox_path" => sandbox_path;
+    force=true,
+)
```
Comment (giordano) [inline deps/build_local_sandbox.jl:16]:
```suggestion
)

```

**Review by giordano — APPROVED** [review]
Looks good

**Comment by maleadt** [comment]
That's awesome, thanks! I was hoping to add some improvements to the userns sandbox, so this come will in handy.

**Pull request merged**